### PR TITLE
Reduce noise associated with `WritersReturnBool` deprecation flag

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
PR: https://github.com/chapel-lang/chapel/pull/20508 introduced a new config-param (`WritersReturnBool`) that allowed users to toggle between implementations of channel writer methods that do/don't return a boolean value.

It defaulted to `true`, maintaining the deprecated behavior (i.e. keep the boolean return value, and emit a warning whenever one of the relevant methods is used). This created too much noise for larger programs, and was unlikely to be helpful given that this return value is rarely used in practice.

This PR does two things to alleviate that noise:
- Switches the default value of `WritersReturnBool` to false.
- Causes the compiler to emit a specialized warning when a users code tries to use the return value from one of the affected methods: `illegal use of boolean return value from 'writef'. Note: this functionality is deprecated; recompile with -sWritersReturnBool=true to continue using the deprecated behavior.`

(it also reformats the normal deprecation warnings to be slightly more helpful)

- [x] paratest passing

### CODE REVIEW

File: compiler/resolution/functionResolution.cpp [hunk 1]
```diff
@@ -8651,10 +8651,35 @@ static Type* moveDetermineRhsTypeErrorIfInvalid(CallExpr* call) {
           if (rhsFn->hasFlag(FLAG_PROMOTION_WRAPPER))
             rhsName = unwrapFnName(rhsFn);
 
-          USR_FATAL(userCall(call),
+          // this deprecation-specific error message should be removed
```
[the remark this answers is not in the document]
Comment (jeremiah-corrado) [reply compiler/resolution/functionResolution.cpp:8654]: Ok should be fixed, thanks for pointing that out!

**Comment by jeremiah-corrado** [comment]
> However, the result feels unnecessarily complex to me. I think it would be enough to change the behavior without any sort of warning or error at all mentioning it was deprecated.

Let me think about this a bit. Arkouda doesn't appear to use return values from any `channel.write*` methods, but I can still imagine someone writing code as if they were checking for early EOF, and not really realizing that the methods always return true. In such cases, it might be nice for them to be able to put off a code correction for later using the param-flag.

Pinging @bradcray to see if he has thoughts on this, since we were discussing this approach offline.

> But even that is arguably more than is needed because these functions would always return true and could never return false, as they were written before PR https://github.com/chapel-lang/chapel/pull/20508, anyway. (Am I wrong about that?).

I believe you are correct. IIRC, I remember reading that these methods were designed with the return value in order to be symmetric with the analogous `channel.read*` methods. (Probably with the idea of making them return false for early EOF sometime in the future?)

File: compiler/resolution/functionResolution.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bradcray) [reply compiler/resolution/functionResolution.cpp:8654]: I was imagining we'd put a temporary pragma on the routines in question that we could check for in here, but didn't want to overwhelm Jeremiah with that idea when originally pitching that he look into this.  That said, I don't have a problem with Michael's approach either, which has the benefit of not requiring the pragma (and the strcmps should only be executed if we're about to error out anyway, so shouldn't slow down compiles?)

**Comment by bradcray** [comment]
I hadn't realized that the return value wasn't actually useful prior to #20508...  If we're confident in that, I tend to agree that  changing it to `void` without a deprecation seems OK in that it's practically like a "fixing a bug" case.  For example, if we think a code wasn't actually getting useful information out of them, it seems unlikely real codes would be relying on them in any practical sense?  (which is to say, they may be trying to store the result, but they can't be relying on that result's value changing...?)  Yes, they'd get a compilation error, but that's a pretty easy change to reply to without a soft transition (as compared to, say, changing the definition of `.high` on a strided range).

If it wasn't for our recent "keep deprecations around six months" conversation, I'd say "Let's just go with these PRs since they're already done," but based on this discussion, I'm not sure I'd want to live with all this deprecation-related code and changes for that long if we don't need to.  Let's bring @lydia-duncan in on this conversation as well since she reviewed #20508.

**Comment by jeremiah-corrado** [comment]
After some offline discussion, we've decided to merge this PR for now, and come back with a second PR that removes the `WritersReturnBool` param and the other deprecation related code.

**Pull request merged**